model-optimizer/mo/front_ng/extractor.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import re
+from collections import defaultdict
+from copy import copy
+
+import numpy as np
+
+from mo.utils.error import Error
+
+
+def fe_decodeNameWithPort(inputModel, node_name: str):
+    """
+    Decode name with optional port specification w/o traversing all the nodes in the graph
+    :param inputModel: Input Model
+    :param node_name:
```

Comment:
add comment describing possible values for this parameter, how the name is formed, what is the delimeter between node name and port.

Reply:
Added TODO related to delimiters between port index, port group name, and node name. It will be needed later, not required in initial support Phase 1